clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2276,6 +2399,11 @@ private Node leader(PartitionInfo partitionInfo) {
         return partitionInfo.leader();
     }
 
+    // This is used in the describe topics path if using DescribeTopicPartitions API.
+    private Node replicaToFakeNode(int id) {
+        return new Node(id, "Dummy", 0);
```

Comment:
Don't we break the contract of the admin api if we do this? The expectation is that the real nodes are returned. We could perhaps get them from the metadata?

Reply:
Good point. Updated.